Automattic/WP-Job-Manager
# Add agreement checkbox to job submission

Fixes #1921

### Changes proposed in this Pull Request

* It introduces a new agreement checkbox to the job submission.
* This new field needs to be enabled through the settings. It's disabled by default.
* The user can also set the terms and conditions page to be linked in the label. Thank you for the suggestion @jom! 🙂

### Testing instructions

* Go to the job submission. Make sure no agreement checkbox appears and you can post a job as previously.
* Go to the WP-admin > Job Listings > Settings > Job Submission, check the "Agreement checkbox" option, and save it.
* Post a new job, but this time make sure you can submit only if you mark the agreement checkbox.
* Go to the WP-admin > Job Listings > Settings > Pages, select pages for terms and conditions, and save it.
* Make sure now the checkbox label contains the pages' links.


### New/Updated Hooks

* New filter `job_manager_agreement_label` to change the agreement label content (change the text, links, and so on).


### Screenshot / Video

<img width="716" alt="Screen Shot 2021-05-28 at 18 32 59" src="https://user-images.githubusercontent.com/876340/120043880-2722b700-bfe3-11eb-8a7a-42b3c6ceec5e.png">
